html/webappapis/scripting/event-loops/shared-between-contexts.html

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html>
+<head>
+<meta charset=utf-8>
+<link rel=help href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loops">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+  <script>
+// 8.1.4.1 Definitions
+//
+// > [...]
+// >
+// > There must be at least one browsing context event loop per user agent, and
+// > at most one per unit of related similar-origin browsing contexts.
+
+async_test(function(t) {
+    new Promise(function(resolve) {
+        var child = window.open('/common/blank.html');
+        child.onload = function() {
+            resolve(child);
+          };
+      })
+      .then(function(child) {
+          var fulfilled = child.Promise.resolve();
+
+          child.close();
+
+          Promise.resolve(fulfilled).then(function() {
```

Comment:
Oh jeez.

What happens if you do `Promise.resolve().then(() => fulfilled.then(() => test code))`, or `setTimeout(() => fulfilled.then(() => test code))`? I'm trying to get a sense if the extra Promise.resolve() just happens to make things wait a bit longer, and that's what causes the failure, or if it's actually a matter of something specific to promises.